data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/validatorInclusion/GetValidatorInclusion.java

```diff
@@ -0,0 +1,312 @@
+                "Request successful",
+                SerializableTypeDefinition.<GetValidatorInclusionResponseData>object()
+                    .name("GetValidatorInclusionResponse")
+                    .withField(
+                        "data",
+                        GetValidatorInclusionResponseData.RESPONSE_DATA,
+                        Function.identity())
+                    .build())
+            .withNotFoundResponse()
+            .withServiceUnavailableResponse()
+            .build());
+    this.chainDataProvider = chainDataProvider;
+    this.spec = spec;
+    this.timeProvider = timeProvider;
+  }
+
+  @OpenApi(
+      path = OAPI_ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get Validator Inclusion",
+      tags = {TAG_TEKU, TAG_EXPERIMENTAL},
+      description =
+          "Returns a per-validator summary of how that validator performed during the current epoch.",
+      pathParams = {
+        @OpenApiParam(name = "epoch", description = "Epoch to get data for"),
+        @OpenApiParam(name = "validator_id", description = "validator id")
+      },
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = {@OpenApiContent(type = JSON, from = GetValidatorResponse.class)}),
+        @OpenApiResponse(status = RES_BAD_REQUEST),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR),
+        @OpenApiResponse(
+            status = RES_SERVICE_UNAVAILABLE,
+            description = "Beacon node is currently syncing.")
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    final int validator = request.getPathParameter(PARAMETER_VALIDATOR_ID).intValue();
+    final UInt64 epoch = request.getPathParameter(EPOCH_PARAMETER);
+    request.respondAsync(
+        chainDataProvider
+            .getValidatorInclusionStateAtEpoch(epoch)
+            .thenApply(
+                maybeStateAndMetadata ->
+                    maybeStateAndMetadata
+                        .map(
+                            stateAndMetadata ->
+                                AsyncApiResponse.respondOk(
+                                    getBalanceFromState(stateAndMetadata.getData(), validator)))
+                        .orElse(AsyncApiResponse.respondServiceUnavailable())));
+  }
+
+  private GetValidatorInclusionResponseData getBalanceFromState(
+      final BeaconState state, final int validator) {
+
+    final UInt64 timeStart = timeProvider.getTimeInMillis();
+    final ValidatorStatuses validatorStatuses =
+        spec.atSlot(state.getSlot()).getValidatorStatusFactory().createValidatorStatuses(state);
+    final UInt64 timeEnd = timeProvider.getTimeInMillis();
+    LOG.debug(
+        "Time taken to compute validator statuses: " + timeEnd.minusMinZero(timeStart) + " ms");
```

Comment:
If this were in chain data provider we could just reuse the method here and then extract the validator we need.